src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public abstract class Figure implements FigureInformation {
+    private static final Random random = new Random();
+    private static final ColorSupplier colorSupplier = new ColorSupplier();
+    private String color = colorSupplier.getRandomColor();
+
+    public abstract Figure createRandomFigure();
```

Comment:
The method `createRandomFigure` should not be in the `Figure` class. Instead, this behavior should be part of a separate class like `FigureSupplier`.